src/rust/scylla-provisioner/src/server.rs

```diff
@@ -0,0 +1,156 @@
+use std::{
+    sync::Arc,
+    time::Duration,
+};
+
+use async_trait::async_trait;
+use rust_proto::{
+    graplinc::grapl::api::scylla_provisioner::v1beta1::{
+        messages as native,
+        server::{
+            ScyllaProvisionerApi,
+            ScyllaProvisionerServer,
+        },
+    },
+    protocol::{
+        healthcheck::HealthcheckStatus,
+        status::Status,
+    },
+};
+use scylla::{
+    transport::errors::QueryError,
+    Session,
+};
+use tokio::net::TcpListener;
+
+use crate::{
+    config::ScyllaProvisionerServiceConfig,
+    table_names::{
+        tenant_keyspace_name,
+        IMM_STRING_TABLE_NAME,
+    },
+};
+
+#[derive(thiserror::Error, Debug)]
+pub enum ScyllaProvisionerError {
+    #[error("Scylla Error {0}")]
+    ScyllaError(#[from] QueryError),
+}
+
+impl From<ScyllaProvisionerError> for Status {
+    fn from(error: ScyllaProvisionerError) -> Self {
+        match error {
+            ScyllaProvisionerError::ScyllaError(error) => {
+                Status::unknown(format!("Scylla Error: {}", error).as_str())
```

Comment:
I'd just call `error.to_string()` instead of using the `format` marco, because it looks like the result of that macro would be "Scylla Error: Scylla Error ...", where those dots are whatever the Display of QueryError would be. Formatting the same thing twice here.

Either that, or just make the error variant #[error(transparent)] and remove the format string from it there. It's not really adding any info there anyway.

Reply:
In general we seem to follow a pattern of making our errors over-the-wire as, well, vague and devoid of detail as possible; including the details of the real error is kind of an aberration here. I'll change it for now, but something to consider.